metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py

```diff
@@ -386,43 +681,65 @@ def construct_job_workunits(
             "kafka-connect", connector_name, self.config.env
         )
 
-        job_property_bag: Optional[Dict[str, str]] = None
-
         lineages = connector.lineages
         if lineages:
             for lineage in lineages:
                 source_dataset = lineage.source_dataset
                 source_platform = lineage.source_platform
                 target_dataset = lineage.target_dataset
                 target_platform = lineage.target_platform
+                # job_property_bag = lineage.job_property_bag
 
-                job_urn = builder.make_data_job_urn_with_flow(flow_urn, source_dataset)
+                job_id = (
+                    source_dataset
+                    if source_dataset
+                    else f"unknown_source.{target_dataset}"
+                )
+                job_urn = builder.make_data_job_urn_with_flow(flow_urn, job_id)
 
-                inlets = [builder.make_dataset_urn(source_platform, source_dataset)]
+                inlets = (
+                    [builder.make_dataset_urn(source_platform, source_dataset)]
+                    if source_dataset
+                    else []
+                )
                 outlets = [builder.make_dataset_urn(target_platform, target_dataset)]
 
-                mce = models.MetadataChangeEventClass(
-                    proposedSnapshot=models.DataJobSnapshotClass(
-                        urn=job_urn,
-                        aspects=[
-                            models.DataJobInfoClass(
-                                name=f"{connector_name}:{source_dataset}",
-                                type="COMMAND",
-                                description=None,
-                                customProperties=job_property_bag,
-                                # externalUrl=job_url,
-                            ),
-                            models.DataJobInputOutputClass(
-                                inputDatasets=inlets or [],
-                                outputDatasets=outlets or [],
-                            ),
-                            # ownership,
-                            # tags,
-                        ],
-                    )
+                mcp = MetadataChangeProposalWrapper(
+                    entityType="dataJob",
+                    entityUrn=job_urn,
+                    changeType=models.ChangeTypeClass.UPSERT,
+                    aspectName="dataJobInfo",
+                    aspect=models.DataJobInfoClass(
+                        name=f"{connector_name}:{job_id}",
+                        type="COMMAND",
+                        description=None,
+                        # customProperties=job_property_bag
```

Comment:
why are we not filling out the `customProperties` anymore?

Reply:
@shirshanka I must have missed to uncomment it. I can fix this, although it would not add any additional information, as there is no custom property in `job_property_bag`. Historically as well, `customProperties` for  `DataJob` was an empty dictionary for kafka connect.